projects/project-player-library/src/lib/shared/task-card/task-card.component.ts

```diff
@@ -23,8 +25,8 @@ export class TaskCardComponent {
   }
 
   moveToDetailsTask(data: any) {
-    if (!this.submittedImprovement) {
-      this.routerService.navigate(`/task-details/${data}`);
+    if (!this.submittedImprovement && !this.startImprovement && (this.task.type !== 'observation')) {
```

Comment:
why this condition and is it a generic condition?

Reply:
no

Reply:
Do make it generic